src/Types/Infer/App.hs

```diff
@@ -0,0 +1,224 @@
+{-# LANGUAGE FlexibleContexts, TupleSections, ScopedTypeVariables,
+   TypeFamilies, CPP, StandaloneDeriving, UndecidableInstances #-}
+module Types.Infer.App (inferApps) where
+
+import Prelude
+
+import qualified Data.Map.Strict as Map
+import qualified Data.Text as T
+import Data.Traversable
+import Data.Foldable
+import Data.Spanned
+import Data.Reason
+import Data.Triple
+import Data.Maybe
+
+import Control.Monad.Infer
+import Control.Lens
+
+import Syntax.Subst
+import Syntax.Var
+import Syntax
+
+import Types.Infer.Builtin
+import {-# SOURCE #-} Types.Infer
+import Types.Kinds
+import Types.Unify
+
+import Text.Pretty.Semantic
+
+
+#ifdef TRACE_TC
+import Debug.Trace
+#endif
+
+-- Infer an application, supporting "Quick Look" impredicativity: We
```

Comment:
Possibly make this `-- |`, so it's a doc comment?

Reply:
done